### ISSUE TITLE
Task Tree with Groups and atomic Skills

### ISSUE THREAD

**Issue by LeanderVonSeelstrang** [RoboCupAtHome/RuleBook#835]
## Is your idea/suggestion related to a problem? Please describe.
This is a written draft as the implementation in Latex turned out to take more time than expected. 

The Score Values and the Group Bonis are not proposed yet. Currently it is only about semantics. 

## Describe the solution you'd like
We would like to gather feedback, if it is worth implementing the Skill Dictionary and Groups in general.

If so we would be interested in potential problems you see with this method. 

Do you think it is understandable?

Do you think it would be doable for the referees?

Do you think it adds value?

[Rulebook dict.pdf](https://github.com/RoboCupAtHome/RuleBook/files/13545970/Rulebook.dict.pdf)
[Rulebook stage 1.pdf](https://github.com/RoboCupAtHome/RuleBook/files/13545971/Rulebook.stage.1.pdf)
[Rulebook stage 2 resize.pdf](https://github.com/RoboCupAtHome/RuleBook/files/13545972/Rulebook.stage.2.resize.pdf)

**Comment by LeanderVonSeelstrang** [comment]
[SkillDictionary.pdf](https://github.com/RoboCupAtHome/RuleBook/files/13702678/SkillDictionary.pdf)

So here is the proposal how it would look like in the Rulebook. 
I wrote several macros:
- you can create a new Skill with \DefSkill{SkillId}{Points}{Default Description}{Category}
- you can render the dictionary \RenderDictionary
- you can render the dictionary by category \RenderCategoryDictionary
- you can use this Skill (similar to citing) with \Skill{SkillId} bzw. \Skill{SkillId}[Sheet specific description]
- you can define Groups with an environment, e.g.
\begin{Group}{GroupId}{Description for the group content}{Bonus Points}{Group Type}
    \Skill{SkillId}  % Reference to a skill
    \GroupRef{otherGroupId} % Reference to another group
\end{Group}
- you can reference a Group with \GroupRef

I am using the \Skill{SkillId} command to render the dictionary, so it is a little bit tricky to label the skills and generate hyperlinks. I might define a helper macro that is just labeling + \Skill{SkillId} which would be used when generating the dictionary, if you think hyperlinks to the skill definition are important.

**Comment by LeanderVonSeelstrang** [comment]
[SkillDictionary_update.pdf](https://github.com/RoboCupAtHome/RuleBook/files/13837976/SkillDictionary_update.pdf)

So here is an update:
* I highlighted the Main Task, so the user knows where to start reading. 
* Skills are now linked to the corresponding skill in the skill dict
* I don't show IDs, because they reduce readability
* I have assigned arbitrary, halfway sensible points to the skills
* I introduced error handling. I am not sure if this is even desired, because now everything compiles without errors, even though some IDs are non-existent. But it is highlighted bright red. 

Feedback would be appreciated.

**Comment by LeanderVonSeelstrang** [comment]
[SkillDictUpdate2.pdf](https://github.com/RoboCupAtHome/RuleBook/files/13848498/SkillDictUpdate2.pdf)

Update 2. Added:
* a motivation for the alternative structure
* explanation how to read it
* explanation and examples how the bonuses work
* section, that human help is not a robot action

Some bonuses are undervalued. Manipulation currently generally scores way more than perception. I never calculated how many points will be realistically achieved in each task (which would be necessary to balance them; I think we still want the tasks to be equally valuable. Not by definition, but by design, though). Also it is possible, that I defined some skills, that are never used.

Next I will integrate these changes into the actual RuleBook and create a PR.
